interpolator/include/interpolator.inc

```diff
@@ -1277,64 +925,64 @@ end subroutine get_axis_level_data
 !! @param [inout] <q> No description
 !! @param [inout] <xs> No description
 !! @param [inout] <ys> No description
- subroutine cart_to_latlon(np, q, xs, ys)
+ subroutine CART_TO_LATLON_(np, q, xs, ys)
 ! vector version of cart_to_latlon1
   integer, intent(in):: np
-  real, intent(inout):: q(3,np)
-  real, intent(inout):: xs(np), ys(np)
+  real(FMS_INTP_KIND_), intent(inout):: q(3,np)
+  real(FMS_INTP_KIND_), intent(inout):: xs(np), ys(np)
 ! local
-  real, parameter:: esl=1.e-10
-  real (f_p):: p(3)
-  real (f_p):: dist, lat, lon
+  real(f_p), parameter:: esl=1.e-10_f_p
```

Comment:
Perhaps the change of f_p should go into main first.  @mcallic2 can f_p be changed with your ongoing work?

Reply:
I would think `f_p` would benefit from having a more descriptive name, but I would agree this would not be a "mixed mode" fix and should be discussed with the fms group. If an issue gets created and it gets the okay, I would be happy to create a pr for it